src/components/pages/listBooksPage/ListBooks.tsx

```diff
@@ -123,6 +123,24 @@ const ListBooks: FC = (): JSX.Element => {
     return isLoaning;
   };
 
+  const updateBook = (book: Book) => {
+    console.log("updating");
+    const index = books.map((e) => e.id).indexOf(book.id);
+    books.splice(index, 1, book);
```

Comment:
Maybe use setBooks instead of mutating the array in the state? 

*"Arrays are mutable in JavaScript, but you should treat them as immutable when you store them in state. Just like with objects, when you want to update an array stored in state, **you need to create a new one (or make a copy of an existing one), and then set state to use the new array.**"*

https://react.dev/learn/updating-arrays-in-state

Maybe also use `Array.findIndex` to find the index https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

Reply:
Function is now updated.